metrics/metrics.go

```diff
@@ -87,14 +89,40 @@ func InitPrometheus(prometheusHost, prometheusPort, prometheusEntry string) {
 	handler := promhttp.HandlerFor(promReg, promhttp.HandlerOpts{EnableOpenMetrics: true})
 	http.Handle("/"+prometheusEntry, handler)
 	go func() {
-		glog.Infof("Starting Prometheus on %v:%v, entry point is /%v", prometheusHost, prometheusPort, prometheusEntry)
-		if err := http.ListenAndServe(prometheusHost+":"+prometheusPort, nil); err != nil {
-			glog.Errorf("Error starting Prometheus at port %v:%v: %v", prometheusHost, prometheusPort, err)
-			os.Exit(1)
+
+		if strings.HasPrefix(prometheusHost, "unix:") {
+			socketPath := prometheusHost[5:] // trim the "unix:" prefix
+			glog.Infof("Starting Prometheus on Unix socket %v, entry point is /%v", socketPath, prometheusEntry)
+			serveOnUnixSocket(socketPath)
+		} else {
+			ipPort := prometheusHost+":"+prometheusPort
+			glog.Infof("Starting Prometheus on IP port %v:%v, entry point is /%v", ipPort, prometheusEntry)
```

Comment:
There are 3 %v but only two arguments provided.

Reply:
fixed in https://github.com/shizunge/endlessh-go/pull/128/commits/a6390a14330a76bab8e31ea1944c6287d7c666ff